contracts/MarginBase.sol

```diff
@@ -82,6 +82,36 @@ contract MarginBase is MinimalProxyable, IMarginBase, OpsReady {
     /// @param amount: amount of token recovered
     event Rescued(address token, uint256 amount);
 
+    /// @notice emitted when an advanced order is placed
+    /// @param account: account placing the order
+    /// @param orderId: id of order
+    /// @param marketKey: futures market key
+    /// @param marginDelta: margin change
+    /// @param sizeDelta: size change
+    /// @param targetPrice: targeted fill price
+    event OrderPlaced(
+        address indexed account,
+        uint256 orderId,
+        bytes32 marketKey,
+        int256 marginDelta,
+        int256 sizeDelta,
+        uint256 targetPrice,
+        OrderTypes orderType
+    );
+
+    /// @notice emitted when an advanced order is cancelled
+    event OrderCancelled(address indexed account, uint256 orderId);
+
+    /// @notice emitted when an advanced order is filled
+    /// @param fillPrice: price the order was executed at
+    /// @param keeperFee: fees paid to the executor
+    event OrderFilled(
+        address indexed account,
+        uint256 orderId,
+        uint256 fillPrice,
+        uint256 keeperFee
+    );
+
```

Comment:
@Tburm let me know if these work for you

Reply:
this is perfect, thanks!